### PR TITLE
Set XDS user agent in ruby via macros

### PR DESCRIPTION
**Pull request by apolcyn**
Uses macros from #26189

### CODE REVIEW

File: src/ruby/ext/grpc/extconf.rb [hunk 1]
```diff
@@ -56,6 +56,8 @@
 end
 
 ENV['CPPFLAGS'] = '-DGPR_BACKWARDS_COMPATIBILITY_MODE'
+ENV['CPPFLAGS'] += ' -DGRPC_XDS_USER_AGENT_SUFFIX=RUBY '
```
Comment (markdroth) [inline src/ruby/ext/grpc/extconf.rb:59]: The values for these macros need to include the quotes.

**Review by markdroth — APPROVED** [review]
I've just merged #26189.  Let's try to get this tested and merged before the branch cut for the next release (next week).  Thanks!

**Comment by apolcyn** [comment]
cc @yashykt @markdroth this PR now also addresses https://github.com/grpc/grpc/pull/26207#discussion_r652165398

**Comment by apolcyn** [comment]
objc ios interop test failure: https://github.com/grpc/grpc/issues/26061

**Pull request merged**